go-controller/pkg/crd/userdefinednetwork/v1/types.go

```diff
@@ -229,3 +284,35 @@ type UserDefinedNetworkStatus struct {
 	// +listMapKey=type
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
+
+// ClusterUserDefinedNetworkStatus contains the observed status of the ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkStatus struct {
+	// ActiveNamespaces indicates the network state in namespaces it available at.
+	// +optional
+	ActiveNamespaces []NamespaceNetworkStatus `json:"activeNamespaces,omitempty"`
```

Comment:
After some discussion, we came up with a status reporting framework described in https://github.com/ovn-org/ovn-kubernetes/pull/4647
For this CRD it means, we can still use only conditions, but report "ActiveNamespaces" via one of the conditions.
The idea is for every condition type, (that probably will be the same for UDN and CUDN like NetworkAttachmentDefinitionReady and NetworkReady in the future) we can report a list of namespaces that were selected together with "Success" status, but if at least one of the namespaces failed, we could report a generic message saying "Network configuration failed for one or more namespaces, check UDN events for more info.", and then report per-namespace error via UDN events.

Reply:
also, looking at conditions themselves, I see it was suggested by @trozet that "NetworkCreated == the network topology has been created in OVN. NetworkReady== the NAD has been created." For me it makes more sense vise versa: NetworkCreated == the NAD has been created NetworkReady== the network topology has been created in OVN. wdyt?

Reply:
ActiveNamespaces are removed form status, in follow up commit status reporting logic is changed to reflect active-namespaces through condition.